packages/server-types/src/types.ts

```diff
@@ -139,12 +139,19 @@ export interface GraphQLRequest {
 
 export type VariableValues = { [name: string]: any };
 
+// Its not possible for this to be Record<string, any> without casting either the
+// result of executeOperation in ApolloServer.ts or the result of sendResponse and
+// sendErrorResponse in requestPipeline.ts.
+// By leaving this open as the wider object type it fulfills what a user can pass
+// in which is an extension of Record<string, any>.
+type GraphQLResponseData = Record<string | number | symbol, any>;
```

Comment:
This can be changed to `Record<string, any>` as outlined in the comment by casting further into the call stack or at the end of the call stack. 

I prefer not to cast so this works for the use case happy to update if you feel otherwise